pennylane/vqe/vqe.py

```diff
@@ -174,29 +174,28 @@ def simplify(self):
           (-1) [X0]
         + (1) [Y2]
         """
-        coeffs = []
-        ops = []
+        ops_coeffs = dict()
 
         for c, op in zip(self.coeffs, self.ops):
-            op = op if isinstance(op, Tensor) else Tensor(op)
-
-            ind = None
-            for i, other in enumerate(ops):
-                if op.compare(other):
-                    ind = i
-                    break
-
-            if ind is not None:
-                coeffs[ind] += c
-                if np.allclose([coeffs[ind]], [0]):
-                    del coeffs[ind]
-                    del ops[ind]
-            else:
-                ops.append(op.prune())
-                coeffs.append(c)
+            op = (op if isinstance(op, Tensor) else Tensor(op)).prune()
+            op_id = frozenset(op._obs_data())
```

Comment:
Do you use this as a key because it is safer than using the operator? Good idea!